msal/src/main/java/com/microsoft/identity/client/AcquireTokenParameters.java

```diff
@@ -0,0 +1,127 @@
+//  Copyright (c) Microsoft Corporation.
+//  All rights reserved.
+//
+//  This code is licensed under the MIT License.
+//
+//  Permission is hereby granted, free of charge, to any person obtaining a copy
+//  of this software and associated documentation files(the "Software"), to deal
+//  in the Software without restriction, including without limitation the rights
+//  to use, copy, modify, merge, publish, distribute, sublicense, and / or sell
+//  copies of the Software, and to permit persons to whom the Software is
+//  furnished to do so, subject to the following conditions :
+//
+//  The above copyright notice and this permission notice shall be included in
+//  all copies or substantial portions of the Software.
+//
+//  THE SOFTWARE IS PROVIDED "AS IS", WITHOUT WARRANTY OF ANY KIND, EXPRESS OR
+//  IMPLIED, INCLUDING BUT NOT LIMITED TO THE WARRANTIES OF MERCHANTABILITY,
+//  FITNESS FOR A PARTICULAR PURPOSE AND NONINFRINGEMENT. IN NO EVENT SHALL THE
+//  AUTHORS OR COPYRIGHT HOLDERS BE LIABLE FOR ANY CLAIM, DAMAGES OR OTHER
+//  LIABILITY, WHETHER IN AN ACTION OF CONTRACT, TORT OR OTHERWISE, ARISING FROM,
+//  OUT OF OR IN CONNECTION WITH THE SOFTWARE OR THE USE OR OTHER DEALINGS IN
+//  THE SOFTWARE.
+
+package com.microsoft.identity.client;
+
+import android.app.Activity;
+import android.util.Pair;
+
+import java.util.List;
+
+/**
+ * Encapsulates the parameters passed to the acquireToken methods of PublicClientApplication
+ */
+public class AcquireTokenParameters extends TokenParameters {
+
+    private Activity mActivity;
+    private String mLoginHint;
+    private UiBehavior mUIBehavior;
+    private List<String> mExtraScopesToConsent;
+    private List<Pair<String, String>> mExtraQueryStringParameters;
+
+
+    /**
+     * Non-null {@link Activity} that will be used as the parent activity for launching the {@link AuthenticationActivity}
+     * @return
+     */
+    public Activity getActivity() {
```

Comment:
Something we might want to consider: if we declare this to be a `Context` we could add support for [`Services`](https://developer.android.com/reference/android/app/Service) to support silent background scenarios

Reply:
Discussed offline